app/services/dsi_client.rb

```diff
@@ -0,0 +1,93 @@
+class DSIClient
+  def initialize(user_id:, organisation_id:)
+    @organisation_id = organisation_id
+    @user_id = user_id
+  end
+
+  def role_ids
+    @role_ids ||= roles.map { |r| r[:id] }
+  end
+
+  def roles
+    user_info[:roles]
+  end
+
+  private
+
+  def user_info
+    @user_info ||= get("/services/#{service_id}/organisations/#{@organisation_id}/users/#{@user_id}")
+  end
+
+  def get(path)
+    uri = uri_for_path(path)
+    request = request_for_uri(uri)
+
+    response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+      http.request(request)
+    end
+
+    DSIClient::Response.new(response).tap do |r|
+      raise DSIClient::RequestInvalid, response.code if r.invalid?
+      raise DSIClient::RequestFailed, response.code if r.failure?
+    end
+  end
+
+  def uri_for_path(path)
+    URI(ENV["DFE_SIGN_IN_URL"]).tap do |uri|
```

Comment:
This and possibly some of the other `ENV.[]` calls in this class might benefit from `ENV.fetch` just for a slightly earlier and more meaningful error in case the env var isn't set